app/services/base_api_manager.rb

```diff
@@ -0,0 +1,66 @@
+require 'restclient'
+
+class BaseAPIManager
+
+  def initialize(organization = "default")
+    @organization = organization
+  end
+
+  #Fetches all the resources for the specific entity
+  def get_entities(entity_name = "")
+    headers = {
+      "Accept" => "application/json",
+      "Authorization" => "Bearer #{organization.oauth_token}"
+    }
+  response = RestClient::Request.execute method: :get, url: "https://api.getbase.com/v2/#{entity_name.downcase.pluralize}", headers: headers
+  #DataParser strips the response from 'items' and 'data' fields
+  DataParser.new.from_base_collection(response)
+  end
+
+  #creates an entity with the parameters passed
+  def create_entities(mapped_connec_entity, external_entity_name)
+    headers = {
+      "Accept" => "application/json",
+      "Content-Type" => "application/json",
+      "Authorization" => "Bearer #{organization.oauth_token}"
+    }
+
+    #DataParser adds the 'data' field before pushing to BaseCRM
+    body = DataParser.new.to_base(mapped_connec_entity)
+    payload = JSON.generate(body)
```

Comment:
A part the fact that JSON.generate takes only objects and arrays they seem to be equivalent. Maybe encapsulate into DataParser?